website/blog/2023-06-04-databend-weekly-96.md

```diff
@@ -0,0 +1,115 @@
+
+select val['a'], val['b'] from test;
++----------+----------+
+| val['a'] | val['b'] |
++----------+----------+
+| 33       | 44       |
+| 55       | 66       |
++----------+----------+
+```
+
+If you are interested in learning more, please check out the resources listed below:
+
+- [RFC | JSON Optimization - Virtual Column](https://databend.rs/doc/contributing/rfcs/json-optimization#virtual-column)
+- [PR | feat(query): Support generate virtual columns](https://github.com/datafuselabs/databend/pull/11590)
+
+## Code Corner
+
+Discover some fascinating code snippets or projects that showcase our work or learning journey.
+
+### Interactive Documents Powered by Askbend
+
+Do you remember [AskBend](https://github.com/datafuselabs/askbend)? It is a Knowledge Base Search and Completion project created and maintained by the Databend team, combining the simplicity of Databend with the magic of OpenAI.
+
+Now, we have integrated it into the official website of Databend, providing an interactive experience for document queries.
+
+![AI Search](../static/img/blog/ai-search.png)
+
+![AI Search Result](../static/img/blog/ai-search-result.png)
+
+If you are interested in learning more, please check out the resources listed below:
+
+- <https://databend.rs>
+- [PR | feat(front search): Add AI searching](https://github.com/datafuselabs/databend/pull/11633)
+
+## Highlights
+
+We have also made these improvements to Databend that we hope you will find helpful:
+
+- Added expression support to Windows function.
+- Added range function & aliases in select list
+- Added support for deduplication of insertion/mutation/update based on tags.
+- Added optimization hint support for COPY INTO.
+- Read *[Docs | Deploying a Databend Cluster](https://databend.rs/doc/deploy/metasrv/metasrv-deploy)* to learn about the tips when manually deploying a Databend cluster.
+- Read *[Docs | Data Types - BITMAP](https://databend.rs/doc/sql-reference/data-types/data-type-bitmap)* and *[Docs | Bitmap Functions](https://databend.rs/doc/sql-functions/bitmap-functions/)* to understand the usage of the BITMAP data type in Databend.
+
+## What's Up Next
+
+We're always open to cutting-edge technologies and innovative ideas. You're more than welcome to join the community and bring them to Databend.
+
+### Use Loser Tree to Optimize Sorting Algorithm
+
+Currently, Databend's sorting algorithm uses MergeSort based on a heap. 
+
+There are three main algorithms for multiple merging: heap sort, winner tree, and loser tree. In all three algorithms, each heap adjustment of the heap sorting requires comparison with the left and right child nodes. The number of comparisons is 2logN while the comparison times for winner tree and loser tree adjustments are logN.
+
+The difference between them is that the winner tree needs to compare with sibling nodes and update the parent node whereas the loser tree only needs to compare with the parent node resulting in fewer memory access times.
```

Comment:
compare what?

Reply:
winner tree and loser tree

Reply:
The contrast lies in how the two trees operate. In a winner tree, each node must be compared with its sibling nodes and then update the parent node. On the other hand, in a loser tree, a node only requires comparison with the parent node, which leads to fewer instances of memory access.